pmd-java/src/main/java/net/sourceforge/pmd/lang/java/rule/bestpractices/UnusedPrivateFieldRule.java

```diff
@@ -121,4 +136,28 @@ private boolean actuallyUsed(List<NameOccurrence> usages) {
     private boolean isOK(String image) {
         return "serialVersionUID".equals(image) || "serialPersistentFields".equals(image) || "IDENT".equals(image);
     }
+
+    /**
+     * Checks whether the given node is annotated with annotation in the set.
+     * The node should be field declaration.
```

Comment:
It would make sense to assert that assumption by changing the type of the parameter of this method to ASTFieldDeclaration and casting at the call site, thus making the cast fail if the assumption is violated. The code would then be self-documented

Reply:
Done.